runtime/src/main/java/org/corfudb/protocols/wireprotocol/NettyCorfuMessageDecoder.java

```diff
@@ -19,7 +23,34 @@
     @Override
     protected void decode(ChannelHandlerContext channelHandlerContext, ByteBuf byteBuf,
                           List<Object> list) throws Exception {
-        list.add(CorfuMsg.deserialize(byteBuf));
+        // Check the type of message based on first byte
+        byte msgMark = byteBuf.readByte();
+
+        if (msgMark == LEGACY_CORFU_MSG_MARK) {
+            list.add(CorfuMsg.deserialize(byteBuf));
+        } else if (msgMark == PROTO_CORFU_REQUEST_MSG_MARK){
+            ByteBufInputStream msgInputStream = new ByteBufInputStream(byteBuf);
+            try {
```

Comment:
Try-with-resource might be a better approach:
`try (ByteBufInputStream msgInputStream = new ByteBufInputStream(byteBuf)) {
`